### PR TITLE
chore: dev cb merge

### PR DESCRIPTION
**Pull request by awsluja**
#### Description of changes
Strategy:
1. checkout cb
2. create branch from cb "dev-cb-merge-jun-3"
3. merge dev
4. resolve conflicts

Passing CI run:
https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/20741/workflows/4eb78493-5e87-49dc-a461-08c55dbaf585

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/publish-codebuild.sh [hunk 1]
```diff
@@ -13,6 +13,11 @@ git fetch --tags https://github.com/aws-amplify/amplify-cli
 function lernaPublishExitOnFailure {
   # exit on failure
   set -e
+  if [[ "$LOCAL_PUBLISH_TO_LATEST" == "true" ]]; then
+    # registy URL update changes .yarnrc.yml file
+    git update-index --assume-unchanged .yarnrc.yml
+  fi
+
```
Comment (awsluja) [inline .circleci/publish-codebuild.sh:20]: I copied this over from the changes made to publish.sh

File: package.json [hunk 2]
```diff
@@ -4,69 +4,71 @@
...
+    "build": "lerna run build",
     "check-version-conflicts:beta": "yarn ts-node ./scripts/check-version.ts beta",
     "check-version-conflicts:release": "yarn ts-node ./scripts/check-version.ts",
-    "clean": "nx reset && nx run-many --target=clean --all && yarn rimraf tsconfig.tsbuildinfo",
-    "cloud-e2e": "source ./scripts/cloud-e2e.sh && cloudE2E",
-    "cloud-e2e-beta": "source ./scripts/cloud-e2e.sh && cloudE2EBeta",
-    "cloud-e2e-local": "source ./scripts/cloud-e2e.sh && cloudE2ELocal",
+    "cloud-e2e-cb": "bash -c 'source ./scripts/cloud-e2e.sh && cloudE2E'",
+    "cloud-e2e-cb-beta": "bash -c 'source ./scripts/cloud-e2e.sh && cloudE2EBeta'",
+    "cloud-e2e-cb-local": "bash -c 'source ./scripts/cloud-e2e.sh && cloudE2ELocal'",
```
Comment (awsluja) [inline package.json:16]: I updated this to match the new bash source syntax

File: package.json [hunk 3]
```diff
@@ -4,69 +4,71 @@
...
+    "smoketest": "lerna run smoketest --output-style=stream-without-prefixes",
+    "smoketest@latest": "CLI_VERSION=latest lerna run smoketest --output-style=stream-without-prefixes",
+    "smoketest@rc": "CLI_VERSION=rc lerna run smoketest --output-style=stream-without-prefixes",
     "split-e2e-tests": "yarn ts-node ./scripts/split-e2e-tests.ts && git add .circleci/config.yml",
     "split-e2e-tests-codebuild": "yarn ts-node ./scripts/split-e2e-tests-codebuild.ts",
-    "test-changed": "nx affected --target=test",
-    "test-ci": "nx run-many --parallel=4 --target=test --all --exclude='@aws-amplify/amplify-provider-awscloudformation' -- --ci --runInBand && nx run-many --target=test --projects='@aws-amplify/amplify-provider-awscloudformation' -- -ci --runInBand",
-    "test": "nx run-many --target=test --all -- --runInBand",
+    "test-changed": "lerna run test --since dev",
+    "test-ci": "lerna run test --concurrency 4 -- --ci -i",
```
Comment (awsluja) [inline package.json:57]: we may need to revisit this and implement the exclusion logic for the amplify-provider-awscloudformation test again, but this time with Lerna

File: shared-scripts.sh [hunk 4]
```diff
@@ -421,7 +410,7 @@ function _integrationTest {
 
     echo "Configuring Amplify CLI"
     yarn rm-dev-link && yarn link-dev && yarn rm-aa-dev-link && yarn link-aa-dev
-    export PATH=$(yarn global bin):$PATH
+    export PATH=$(pwd)/.bin:$PATH
```
Comment (awsluja) [inline shared-scripts.sh:413]: this also might break the integration test and need revisited

**Review by jhockett — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

**Pull request merged**